From: Alexander Burmashev <alexander.burmashev@oracle.com>
Date: Thu, 25 Feb 2021 07:42:04 -0800
Subject: [PATCH] Fix build on older gnu-efi

Signed-off-by: Alex Burmashev <alexander.burmashev@oracle.com>
Signed-off-by: Ilya Okomin <ilya.okomin@oracle.com>
diff -Nur shim-15.3.old/csv.c shim-15.3/csv.c
--- shim-15.3.old/csv.c	2021-02-24 20:04:11.000000000 -0800
+++ shim-15.3/csv.c	2021-02-25 03:05:51.351690308 -0800
@@ -21,8 +21,8 @@
 			valid = strntoken(next, max, delims, &token, &state);
 		}
 		if (valid) {
-			next += strlena(token) + 1;
-			max -= strlena(token) + 1;
+			next += strlena((const CHAR8 *)token) + 1;
+			max -= strlena((const CHAR8 *)token) + 1;
 			columns[n] = token;
 			new_n = n + 1;
 		} else {
@@ -60,7 +60,7 @@
 
 	max = (uintptr_t)end - (uintptr_t)line + (end > line ? 1 : 0);
 
-	if (line && is_utf8_bom(line, max))
+	if (line && is_utf8_bom((CHAR8 *)line, max))
 		line += UTF8_BOM_SIZE;
 
 	while (line && line <= data_end) {
diff -Nur shim-15.3.old/include/sbat.h shim-15.3/include/sbat.h
--- shim-15.3.old/include/sbat.h	2021-02-24 20:04:11.000000000 -0800
+++ shim-15.3/include/sbat.h	2021-02-25 03:06:35.567417137 -0800
@@ -9,12 +9,12 @@
 extern UINTN _sbat, _esbat;
 
 struct sbat_var_entry {
-	const CHAR8 *component_name;
-	const CHAR8 *component_generation;
+	const char *component_name;
+	const char *component_generation;
 	/*
 	 * This column is only actually on the "sbat" version entry
 	 */
-	const CHAR8 *sbat_datestamp;
+	const char *sbat_datestamp;
 	list_t list;
 };
 extern list_t sbat_var;
@@ -25,12 +25,12 @@
 void cleanup_sbat_var(list_t *entries);
 
 struct sbat_section_entry {
-	const CHAR8 *component_name;
-	const CHAR8 *component_generation;
-	const CHAR8 *vendor_name;
-	const CHAR8 *vendor_package_name;
-	const CHAR8 *vendor_version;
-	const CHAR8 *vendor_url;
+	const char *component_name;
+	const char *component_generation;
+	const char *vendor_name;
+	const char *vendor_package_name;
+	const char *vendor_version;
+	const char *vendor_url;
 };
 #define SBAT_SECTION_COLUMNS (sizeof (struct sbat_section_entry) / sizeof(CHAR8 *))

 EFI_STATUS
diff -Nur shim-15.3.old/sbat.c shim-15.3/sbat.c
--- shim-15.3.old/sbat.c	2021-02-24 20:04:11.000000000 -0800
+++ shim-15.3/sbat.c	2021-02-25 03:07:38.768454862 -0800
@@ -49,7 +49,7 @@ parse_sbat_section(char *section_base, size_t section_size,
 				efi_status = EFI_INVALID_PARAMETER;
 				goto err;
 			}
-			allocsz += strlena(row->columns[i]) + 1;
+			allocsz += strlena((const CHAR8 *)row->columns[i]) + 1;
 		}
 		n++;
 	}
@@ -227,7 +227,7 @@ parse_sbat_var_data(list_t *entry_list, UINT8 *data, UINTN datasize)
 				efi_status = EFI_INVALID_PARAMETER;
 				goto err;
 			}
-			allocsz += strlena(row->columns[i]) + 1;
+			allocsz += strlena((const CHAR8 *)row->columns[i]) + 1;
 		}
 		n++;
 	}
